content/src/content/jcr_root/apps/core/wcm/components/commons/datalayer/clientlibs/js/adobe-client-data-layer.min.js

```diff
@@ -0,0 +1,2 @@
ct.keys(e);if(n.length>r.length)return!0;for(var o=0;o<n.length;o++){for(var a=n[o],i=!1,u=0;u<r.length;u++){if(a===r[u]){i=!0;break}}if(!i)return!0}return!1}};e.exports={item:r,utils:f}},{"./DataLayerConstants":7}],9:[function(t,e,n){var r=t("./DataLayerConstants"),o=function(){function e(t){_classCallCheck(this,e),this._event=t.config.on?t.config.on:t.config.off,this._handler=t.config.handler?t.config.handler:null,this._scope=t.config.scope?t.config.scope:null,t.config.on&&null===this._scope&&(this._scope=r.listenerScope.FUTURE),this._path=t.config.path?t.config.path:null}return _createClass(e,[{key:"event",get:function(){return this._event}},{key:"handler",get:function(){return this._handler}},{key:"scope",get:function(){return this._scope}},{key:"path",get:function(){return this._path}}]),e}();e.exports=o},{"./DataLayerConstants":7}],10:[function(t,e,n){var p=t("./DataLayerConstants"),o=t("lodash.has"),_=t("lodash.get"),y=t("lodash.isequal"),h=t("lodash.clonedeep"),r={};r.create=function(s){var f={};function u(t,e,n){if(function(t,e){var n=t.event,r=e.config,o=!1;e.type===p.itemType.DATA?n===p.dataLayerEvent.CHANGE&&(o=l(t,e)):e.type===p.itemType.EVENT&&(n!==p.dataLayerEvent.EVENT&&n!==r.event||(o=l(t,e)),r.data&&n===p.dataLayerEvent.CHANGE&&(o=l(t,e)));return o}(t,e)){var r=e.config,o=h(r);if(e.config.data)if(t.path){var a=_(s._previousStateCopy,t.path),i=_(o.data,t.path);t.handler.call(s._dataLayer,o,a,i)}else if(n)t.handler.call(s._dataLayer,o);else{var u=s._previousStateCopy,c=h(s._state);t.handler.call(s._dataLayer,o,u,c)}else t.handler.call(s._dataLayer,o)}}function l(t,e){var n=e.config;return!t.path||!n.data||o(n.data,t.path)}function r(t){var e=t.event;if(Object.prototype.hasOwnProperty.call(f,e)){var n=!0,r=!1,o=void 0;try{for(var a,i=f[e].entries()[Symbol.iterator]();!(n=(a=i.next()).done);n=!0){var u=_slicedToArray(a.value,2),c=u[0],s=u[1];if(y(s,t))return c}}catch(t){r=!0,o=t}finally{try{n||null==i.return||i.return()}finally{if(r)throw o}}}return-1}return{register:function(t){var e=t.event;Object.prototype.hasOwnProperty.call(f,e)?-1===r(t)&&f[t.event].push(t):f[t.event]=[t]},unregister:function(t){var e=t.event;if(Object.prototype.hasOwnProperty.call(f,e))if(t.handler||t.scope||t.path){var n=r(t);-1<n&&f[e].splice(n,1)}else f[e]=[]},triggerListeners:function(i){(function(t){var e=[],n=t.config;t.type===p.itemType.DATA?e.push(p.dataLayerEvent.CHANGE):t.type===p.itemType.EVENT&&(n.event!==p.dataLayerEvent.CHANGE&&e.push(n.event),e.push(p.dataLayerEvent.EVENT),n.data&&e.push(p.dataLayerEvent.CHANGE));return e})(i).forEach(function(t){if(Object.prototype.hasOwnProperty.call(f,t)){var e=!0,n=!1,r=void 0;try{for(var o,a=f[t][Symbol.iterator]();!(e=(o=a.next()).done);e=!0){u(o.value,i,!1)}}catch(t){n=!0,r=t}finally{try{e||null==a.return||a.return()}finally{if(n)throw r}}}})},triggerListener:function(t,e){u(t,e,!0)}}},e.exports=r},{"./DataLayerConstants":7,"lodash.clonedeep":1,"lodash.get":2,"lodash.has":3,"lodash.isequal":4}]},{},[6]);
```

Comment:
This should be delivered by a launch extension, but it's here currently for the PoC

Reply:
I think that the data layer script should be on a CDN and that we should stay flexible and offer both options for including it on the page:
- through the page component if the data layer is enabled (default)
- though a launch extension
Some customers might want to use the data layer but might not want to use launch.